_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,649 @@
+
+The following request adds twelve hotel documents that contain criteria such as feedback ratings and whether or not parking is available.  
+
+```json
+POST /_bulk
+{ "index": { "_index": "hotels-index", "_id": "1" } }
+{ "location": [5.2, 4.4], "parking" : "true", "rating" : 5 }
+{ "index": { "_index": "hotels-index", "_id": "2" } }
+{ "location": [5.2, 3.9], "parking" : "false", "rating" : 4 }
+{ "index": { "_index": "hotels-index", "_id": "3" } }
+{ "location": [4.9, 3.4], "parking" : "true", "rating" : 9 }
+{ "index": { "_index": "hotels-index", "_id": "4" } }
+{ "location": [4.2, 4.6], "parking" : "false", "rating" : 6}
+{ "index": { "_index": "hotels-index", "_id": "5" } }
+{ "location": [3.3, 4.5], "parking" : "true", "rating" : 8 }
+{ "index": { "_index": "hotels-index", "_id": "6" } }
+{ "location": [6.4, 3.4], "parking" : "true", "rating" : 9 }
+{ "index": { "_index": "hotels-index", "_id": "7" } }
+{ "location": [4.2, 6.2], "parking" : "true", "rating" : 5 }
+{ "index": { "_index": "hotels-index", "_id": "8" } }
+{ "location": [2.4, 4.0], "parking" : "true", "rating" : 8 }
+{ "index": { "_index": "hotels-index", "_id": "9" } }
+{ "location": [1.4, 3.2], "parking" : "false", "rating" : 5 }
+{ "index": { "_index": "hotels-index", "_id": "10" } }
+{ "location": [7.0, 9.9], "parking" : "true", "rating" : 9 }
+{ "index": { "_index": "hotels-index", "_id": "11" } }
+{ "location": [3.0, 2.3], "parking" : "false", "rating" : 6 }
+{ "index": { "_index": "hotels-index", "_id": "12" } }
+{ "location": [5.0, 1.0], "parking" : "true", "rating" : 3 }
+```
+
+#### Sample response
+
+Upon success, you should receive "200-OK" status with entries for each document ID added to the index. The following response is truncated to only show one document:
+
+```json
+{
+  "took" : 140,
+  "errors" : false,
+  "items" : [
+    {
+      "index" : {
+        "_index" : "hotels-index",
+        "_id" : "1",
+        "_version" : 2,
+        "result" : "updated",
+        "_shards" : {
+          "total" : 1,
+          "successful" : 1,
+          "failed" : 0
+        },
+        "_seq_no" : 12,
+        "_primary_term" : 3,
+        "status" : 200
+      }
+    }
+  ]
+}
+
+```
+
+## Step 3: Search your data with a filter
+
+Now you can create a k-NN search that specifies filters using Query DSL Boolean clauses. You need to include your reference point to search for nearest neighbors. Provide an x-y coordinate for the point with the `vector` field, such as `"vector": [ 5.0, 4.0]`
+
+ To learn more about how to specify ranges with Query DSL, see [Range query]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/term/#range-query).
+{: .note }
+
+#### Sample request
+
+The following request creates a k-NN query that only returns the top hotels rated between 8 and 10 and that provide parking. The filter criteria is indicated with the Query DSL `range` query clause to indicate the range for the feedback ratings and a `term` query clause to indicate "parking."
```

Comment:
Would "...the top hotels rated between 8 and 10 that also provide parking" work? Also, I would probably try to avoid "is indicated...to indicate".